packages/davinci-client/src/lib/collector.types.ts

```diff
@@ -43,10 +48,47 @@ export interface SingleValueCollectorNoValue<T extends SingleValueCollectorTypes
   };
 }
 
+/**
+ * Type to help infer the collector based on the collector type
+ * Used specifically in the returnSingleValueCollector wrapper function.
+ * When given a type, it can narrow which type it is returning
+ *
+ * Note: You can see this type in action in the test file or in the collector.utils file.
+ */
+export type InferSingleValueCollectorFromSingleValueCollectorType<
+  T extends SingleValueCollectorTypes,
+> = T extends 'TextCollector'
+  ? TextCollector
+  : T extends 'DropDownCollector'
+    ? DropDownCollector
+    : T extends 'ComboboxCollector'
+      ? ComboboxCollector
+      : T extends 'RadioCollector'
+        ? RadioCollector
+        : //: T extends 'LabelCollector'
+          //? LabelCollector
+          T extends 'PasswordCollector'
+          ? PasswordCollector
+          : T extends 'FlowLinkCollector'
+            ? FlowLinkCollector
+            : /**
+               * At this point, we have not passed in a collector type
+               * or we have explicitly passed in 'SingleValueCollector'
+               * So we can return either a SingleValueCollector with value or without value
+               * or without a value.
+               **/
+              | SingleValueCollectorWithValue<'SingleValueCollector'>
+                | SingleValueCollectorNoValue<'SingleValueCollector'>;
```

Comment:
I'm not sure what to think of this. Is this a recommended pattern that I'm just unfamiliar with, or is there some requirement we have that necessitates such a complex, deeply nested structure?

Reply:
The reason i created this type was to better infer the type.

I definitely do not claim this to be the best way to do it, and if you feel there is not enough value derived from this, we can ignore it.

But i'd like to hear your thoughts outside of just purely this type, if you think the value it provides is worth the "complexity" of the type. (its just how we can write "if else" in types)